pkg/api/options.go

```diff
@@ -34,3 +38,32 @@ func FullObjects(fullObjects bool) ListOption {
 func AutoTag(tags ...string) CreateOption {
 	return internal.AutoTagOption(tags)
 }
+
+// EnvironmentOption can be used to configure an alternative environment path
+// segment for a given API group
+type EnvironmentOption struct {
+	APIGroup       string
+	EnvPathSegment string
+	Override       bool
+}
+
+// ApplyToAny applies the EnvironmentOption to any request type
+func (o EnvironmentOption) ApplyToAny(opts types.Options) error {
+	return opts.Set(fmt.Sprintf("environment/%s", o.APIGroup), o.EnvPathSegment, o.Override)
+}
+
+// GetEnvironmentPathSegment retrieves the environment path segment of a given API group
+// or the provided defaultValue if no environment override is set
+func GetEnvironmentPathSegment(ctx context.Context, apiGroup, defaultValue string) (string, error) {
+	if options, err := types.OptionsFromContext(ctx); err != nil {
+		return "", err
```

Comment:
shouldn't it just return the default value in this case?

Reply:
Hmm.. not sure about that. I think Options are always expected to be available on the context when using this helper (`GetEnvironmentPathSegment`) defaulting to prod (in most cases) on error sounds a bit dangerous.

Reply:
really not sure .. I think the risk to default to prod due to this is pretty low and can be reasonably enough tested with unit tests, but makes the code using this function a lot nicer in exchange

And dealing with the risk is also possible by using tokens that cannot use the wrong service